src/app/modules/form/admin-form/admin-form.controller.ts

```diff
@@ -1449,3 +1450,83 @@ export const handleUpdateFormField = [
   ),
   _handleUpdateFormField,
 ]
+
+/**
+ * NOTE: Exported for testing.
+ * Private handler for POST /forms/:formId/fields
+ * @precondition Must be preceded by request validation
+ * @security session
+ *
+ * @returns 200 with created form field
+ * @returns 403 when current user does not have permissions to create a form field
+ * @returns 404 when form cannot be found
+ * @returns 410 when creating form field for an archived form
+ * @returns 413 when creating form field causes form to be too large to be saved in the database
+ * @returns 422 when an invalid form field creation is attempted on the form
+ * @returns 422 when user in session cannot be retrieved from the database
+ * @returns 500 when database error occurs
+ */
+export const _handleCreateFormField: RequestHandler<
+  { formId: string },
+  FormFieldDto | ErrorDto,
+  FieldCreateDto
+> = (req, res) => {
+  const { formId } = req.params
+  const sessionUserId = (req.session as Express.AuthedSession).user._id
+
+  // Step 1: Retrieve currently logged in user.
+  return (
+    UserService.getPopulatedUserById(sessionUserId)
+      .andThen((user) =>
+        // Step 2: Retrieve form with write permission check.
+        AuthService.getFormAfterPermissionChecks({
+          user,
+          formId,
+          level: PermissionLevel.Write,
+        }),
+      )
+      // Step 3: User has permissions, proceed to create form field with provided body.
+      .andThen((form) => AdminFormService.createFormField(form, req.body))
+      .map((createdFormField) =>
+        res.status(StatusCodes.OK).json(createdFormField),
+      )
+      .mapErr((error) => {
+        logger.error({
+          message: 'Error occurred when creating form field',
+          meta: {
+            action: '_handleCreateFormField',
+            ...createReqMeta(req),
+            userId: sessionUserId,
+            formId,
+            createFieldBody: req.body,
+          },
+          error,
+        })
+        const { errorMessage, statusCode } = mapRouteError(error)
+        return res.status(statusCode).json({ message: errorMessage })
+      })
+  )
+}
+
+/**
+ * Handler for POST /forms/:formId/fields
+ */
+export const handleCreateFormField = [
+  celebrate({
+    [Segments.BODY]: Joi.object({
+      // Ensures id is not provided.
+      _id: Joi.any().forbidden(),
+      globalId: Joi.any().forbidden(),
+      fieldType: Joi.string()
+        .valid(...Object.values(BasicField))
+        .required(),
+      title: Joi.string().required(),
+      description: Joi.string().allow(''),
+      required: Joi.boolean(),
+      disabled: Joi.boolean(),
```

Comment:
not sure but should these be required?

Reply:
These params have a model default already (yea, probably can set the update field one to not be required too)